project/src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -73,4 +81,24 @@ private User getUser(Connection conn, String email) throws SQLException {
       return null;
     }
   }
+
+  private User addUser(Connection conn, String name, String email) throws SQLException {
+    String stmt = "INSERT INTO users (name, email) VALUES ('" + name + "', '" + email + "');";
+    try (PreparedStatement userStmt = conn.prepareStatement(stmt);) {
+      // Execute the statement
+      userStmt.execute();
+      LOGGER.log(Level.INFO, "User " + name + " added to database.");
+
+    }
+    String getId = "SELECT LAST_INSERT_ID();";
+    try (PreparedStatement idStmt = conn.prepareStatement(getId);) {
```

Comment:
when Sgonks becomes very popular, this may cause an issue when 2 new users sign up "instantaneously" could last_insert_id be wrong?

Reply:
https://docs.oracle.com/javadb/10.8.3.0/ref/crefjavstateautogen.html may be of use

Reply:
@leonnemets https://dev.mysql.com/doc/refman/8.0/en/information-functions.html#function_last-insert-id LAST_INSERT_ID operates on a per connection basis so I don't think should be an issue?